pkg/registry/apis/provisioning/jobs/pullrequest/changes.go

```diff
@@ -158,11 +128,13 @@ func calculateFileChangeInfo(ctx context.Context, repo repository.Reader, baseUR
 	err = info.Parsed.DryRun(ctx)
 	if err != nil {
 		info.Error = err.Error()
-		return info, nil
+		return info
 	}
 
 	// Dashboards get special handling
 	if info.Parsed.GVK.Kind == dashboardKind {
+		// FIXME: extract the logic out of a dashboard URL builder/injector or similar
```

Comment:
I refrained myself from jumping into this change now because I want to focus on testing other code paths related to sync / export.